arangod/Aql/Ast.cpp

```diff
@@ -1124,6 +1124,10 @@ AstNode* Ast::createNodeValueInt(int64_t value) {
 
 /// @brief create an AST double value node
 AstNode* Ast::createNodeValueDouble(double value) {
+  if (value == -0.0) {
+    // unify -0.0 and +0.0 
+    value = 0.0;
+  }
```

Comment:
This is inconsistent with the `AqlValueHintDouble` constructor which maps NaN and infinity to null. Is this intentional? If we have to support doubles with NaN values, this complicates comparison if we want to comply to the IEEE 754 standard.
But if we do want to support NaNs, we should at least make sure that we don't have signaling NaNs to avoid invalid operation exceptions!

Reply:
Fixed